### PR TITLE
Nf gems2 python packaging

### PR DESCRIPTION
**Pull request by orwonthe**
Add scripts to build and install gems2Python and samseg wheels

### CODE REVIEW

File: GEMS2/setup.py [hunk 1]
```diff
@@ -3,31 +3,41 @@
...
+
     def has_ext_modules(self):
         return True
 
+
 def create_init_file():
-    file_path = os.path.join(os.path.dirname(__file__), 'bin','__init__.py')
+    file_path = os.path.join(os.path.dirname(__file__), 'bin', '__init__.py')
     if not os.path.exists(file_path):
         with open(file_path, 'w'):
```
Comment (yujanshrestha) [inline GEMS2/setup.py]: FYI this is a potential race condition. A separate process could create the file right after the `if` statement. It is usually better to write (or create a utility method for)
```
try:
    open(file_path, 'r')
except IOError:
    open(file_path, 'w')
```

File: samseg/samseg/show_figures.py [hunk 2]
```diff
@@ -1,11 +1,16 @@
...
 import numpy as np
-import pyqtgraph as pg
-from PyQt5 import QtGui
+try:
+    import pyqtgraph as pg
+    from PyQt5 import QtGui
+
+    from samseg.hdav import hdav
+    from samseg.hdav.hdav.hdav import HdavWindow
+    VISUALIZATION_NOT_INSTALLED = False
```
Comment (yujanshrestha) [inline samseg/samseg/show_figures.py]: For this situation, I want you to be aware of `try`, `except`, `else`. I think it is more clear than relying on the fact the `try` gets to the end.

```python
try
    ... stuff under test ...
except:
    VISUALIZATION_NOT_INSTALLED = True
else:
    VISUALIZATION_NOT_INSTALLED = False
```

File: samseg/samseg/show_figures.py [hunk 3]
```diff
@@ -63,185 +71,189 @@ def start_movie(self, **kwargs):
...
-        if self.show_flag:
-            pg.plot(y=data, pen=pg.mkPen(color='l', width=3), title=title)
-            if self.interactive:
-                QtGui.QApplication.processEvents()
+            if alpha is None:
+                cmap = None
             else:
-                QtGui.QApplication.exec_()
-
+                start = min([np.min(image) for image in image_list])
```
Comment (yujanshrestha) [inline samseg/samseg/show_figures.py:158]: note the `[` is not necessary here. Without the `[]`, this becomes a generator expression which is more memory efficient (not a problem here at all however)

File: samseg/samseg/show_figures.py [hunk 4]
```diff
@@ -63,185 +71,189 @@ def start_movie(self, **kwargs):
...
+                    start=start,
+                    stop=stop,
+                    bottom_alpha=alpha,
+                    top_alpha=alpha)
+            return [{
+                'name': names[index],
+                'data': image_list[index],
+                'visible': visibility,
+                'cmap': cmap,
+            } for index in range(layer_count)]
```
Comment (yujanshrestha) [inline samseg/samseg/show_figures.py]: this can probably replaced with a `zip`

File: samseg/setup.py [hunk 5]
```diff
@@ -15,18 +16,26 @@
...
-    'nibabel >= 2.2.1'
+    'nibabel >= 2.2.1',
     'numpy >= 1.13.3',
     'scipy >= 1.0.0',
     'gems2python',
 ]
 
+PACKAGES_TO_FIND = ['samseg', 'samseg.dev_utils']
+NO_VISUALIZATION_ARG = '--no-visualization'
+if not NO_VISUALIZATION_ARG in sys.argv:
```
Comment (yujanshrestha) [inline samseg/setup.py]: note that you can also do `if NO_VISUALIZATION_ARG not in sys.argv:`

**Review by yujanshrestha — COMMENTED** [review]
Looks great. Very thorough. Minor comments and I think you should add some documentation to a readme file.

File: GEMS2/setup.py (hunk 1 above)
Comment (orwonthe) [reply GEMS2/setup.py]: Good form, I am changed to use it.  I also clarified what the code is trying to do. Might not be obvious that an empty __init__.py file is the intended outcome.

File: samseg/samseg/show_figures.py (hunk 2 above)
Comment (orwonthe) [reply samseg/samseg/show_figures.py]: Good form. I am using it.

File: samseg/setup.py (hunk 5 above)
Comment (orwonthe) [reply samseg/setup.py]: Yes. Reads more naturally that way.

**Pull request merged**